pkg/onchain/sc.go

```diff
@@ -103,57 +113,62 @@ func DeploySC(client *node.Client,
 		signer,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("calling executeSC: %w", err)
+		return nil, nil, fmt.Errorf("calling executeSC: %w", err)
 	}
 
-	eventFound, operationWithEventResponse, err := listenEvents(client, operationResponse)
-	if eventFound {
-		return operationWithEventResponse, err
+	events, err := listenEvents(client, operationResponse.OperationID)
+	if err != nil {
+		return nil, nil, err
 	}
 
-	// If no event received, return a message to announce sc is deployed
-	return &OperationWithEventResponse{
-		Event:             "sc deployed successfully but no event received",
-		OperationResponse: *operationResponse,
-	}, nil
+	return operationResponse, events, nil
 }
 
 func listenEvents(
 	client *node.Client,
-	operationResponse *sendOperation.OperationResponse,
-) (bool, *OperationWithEventResponse, error) {
+	operationID string,
+) ([]node.Event, error) {
 	counter := 0
 
 	ticker := time.NewTicker(time.Second * evenHeartbeat)
 
 	for ; true; <-ticker.C {
 		counter++
 
-		if counter > maxWaitingTimeInSeconds*evenHeartbeat {
+		if counter > maxWaitingTimeInSeconds/evenHeartbeat {
```

Comment:
What is the point of dividing by 1 ?

Reply:
If we change the polling interval constant (evenHeartbeat) this code will still work